modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/maven/AdditionalPluginDependenciesResolver.java

```diff
@@ -159,12 +166,20 @@ private void addPluginDependenciesAdditionalLibraries(List<BundleDependency> app
   protected Collection<Plugin> resolveAdditionalDependenciesFromMulePlugins(List<BundleDependency> mulePlugins) {
     Map<String, Plugin> additionalDependenciesFromMulePlugins = new HashMap<>();
 
+    // See LightweightDeployableProjectModelBuilderTestCase#createDeployableProjectModelWithAdditionalDependenciesInAPlugin
     mulePlugins.forEach(mulePlugin -> {
       try {
-        Model pomModel =
-            aetherMavenClient.getEffectiveModel(toFile(mulePlugin.getBundleUri().toURL()), of(temporaryFolder));
 
-        Build build = pomModel.getBuild();
+        BundleDescriptor descriptor = mulePlugin.getDescriptor();
+        ArtifactCoordinates artifactCoordinates =
+            new ArtifactCoordinates(descriptor.getGroupId(), descriptor.getArtifactId(), descriptor.getVersion());
+        URL mulePluginUrl = mulePlugin.getBundleUri().toURL();
+
+        Supplier<Model> pomModel =
+            pomModels.getOrDefault(artifactCoordinates, () -> aetherMavenClient.getEffectiveModel(toFile(mulePluginUrl),
```

Comment:
can we get the file from the uri instead of from the url? url must be avoided where possible...